ocr_test.go

```
@@ -0,0 +1,268 @@
+
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				// Set global limitOcrPages
+				limitOcrPages = tc.limitPages
+
+				// Make our safety check testable without the full OCR pipeline
+
+				// We can examine the code flow by checking if app.localPDFPath changes
+				// Clear previous output
+				os.RemoveAll(tempPDFDir)
+				os.MkdirAll(tempPDFDir, 0755)
```

golangci-lint (1.64.8)

Error return value of `os.MkdirAll` is not checked

(errcheck)